packages/splitio-adapter/modules/adapter/adapter.spec.js

```diff
@@ -100,15 +88,18 @@ describe('when configuring', () => {
 
   describe('when ready', () => {
     let factory;
+    let onStub;
     let onStatusStateChange;
     let onFlagsStateChange;
 
     beforeEach(() => {
       onStatusStateChange = jest.fn();
       onFlagsStateChange = jest.fn();
+      onStub = jest.fn((_, cb) => cb());
+
       factory = {
         client: jest.fn(() => ({
-          on: jest.fn((_, cb) => cb()),
+          on: onStub,
```

Comment:
Otherwise the the `splitio()` call will generate through `client()` a new `on` stub each time. Especially down in the previous expectation of `client().on` which then has a zero'ed call count.

Reply:
That's what I didn't succeed to do. Nice spot :)